packages/scoped-elements/html-element.js

```diff
@@ -57,7 +57,19 @@ const ScopedElementsMixinImplementation = superclass =>
      */
     attachShadow(options) {
       const { scopedElements } = /** @type {typeof ScopedElementsHost} */ (this.constructor);
-      if (!this.registry) {
+
+      const shouldCreateRegistry =
+        !this.registry ||
+        // @ts-ignore
+        (this.registry === this.constructor.__registry &&
+          !Object.prototype.hasOwnProperty.call(this.constructor, '__registry'));
+
+      /**
+       * Create a new registry if:
+       * - the registry is not defined
+       * - this class doesn't have its own registry *AND* has no shared registry
```

Comment:
can we add to this comment that this is important specifically for superclasses/inheritance?